### PR TITLE
Create Release Notes for Kyma 1.21 Berlin

### PR DESCRIPTION
**Pull request by majakurcius**
**Description**

Create release notes for Kyma 1.21 Berlin. 

Changes proposed in this pull request:

- Add Release Notes for Kyma 1.21

**Related issue(s)**
#683

### CODE REVIEW

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+For more information about derived fields, read the [Grafana documentation](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
+
+## Eventing
+
+### Knative-based Eventing replaced by NATS 
+
+In this release, we removed Knative-based Eventing. 
+The new Eventing is based on [NATS](https://github.com/nats-io/nats-server) with a Kyma-specific Subscription custom resource (CR). 
+For upgrade scenarios, refer to the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.21/docs/migration-guides/1.20-1.21.md).
+For more information on the new Eventing, read the [documentation](https://kyma-project.io/docs/master/components/eventing).
```
Comment (kazydek) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:68]: ## Serverless

### Node.js 10 deprecation

The base image for [Node.js 10 will soon be deprecated](https://nodejs.org/en/about/releases/). Thus, following the recommendations from Node.js, we are planning to remove Node.js 10 from the list of supported runtimes as of the next release. This basically means that Node.js 10-based Functions will continue to work, but you won’t be able to create a new Function on Node.js 10 or edit the existing Function to run on it. 

For now, we recommend you configure all your Node.js 10 Functions to run on the Node.js 12 runtime. For details, see the link to the [document describing incompatibilities](https://nodejs.org/tr/blog/uncategorized/10-lts-to-12-lts/) between the two versions.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- [Monitoring](#monitoring) - dynamic datasource configuration, deriving traces from Loki logs
+- [Eventing](#eventing) - Knative-based Eventing replaced by NATS 
+
+## CLI
+
+### New alpha commands stabilized 
+
+We've recently added new alpha commands: `alpha provision k3s`, `alpha deploy`, `alpha delete`, and  `alpha version`.
+In this release, we stabilized these commands, and they are now ready for beta-testing.
+For more information, read the [Alpha Command Usage Examples](https://kyma-project.io/docs/master/cli/overview#details-kyma-cli-alpha-command-usage-examples) and the command help.
```
Comment (alexandra-simeonova) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:37]:
```suggestion
For more information, read the [Alpha Command Usage Examples](https://kyma-project.io/docs/1.21/cli/overview#details-kyma-cli-alpha-command-usage-examples) and the command help. 
```
In the section [below](https://github.com/kyma-project/website/pull/686/files#diff-40c745baf7c2b64a5870600a8df8a938fef65ca0204286968eee476378ac2cc4R45), we use the 1.21 link, but master works too. It's good if they are consistent though.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+We left the windy Amsterdam, but we're staying within the area of the West Germanic group of languages, this time stopping at Berlin, the capital of Germany and a world city of culture, politics, media, and science. 
+But don't be fooled by the great historical buildings, universities, and museums. 
+Berlin does a great job at combining this rich heritage with the new and avant-garde, and it does all that ahead of the rest, too. 
+Contemporary art, numerous festivals, artisanal coffee shops, craft beer, and the famous nightlife are just part of it. 
+Just like Berlin, Kyma also combines a solid foundation with innovation, providing high quality that will not cost you an arm and a leg. 
+This time, we focused on CLI, Monitoring, and Eventing, so grab a cup of (artisanal) coffee, and read on.
+
+<!-- overview -->
+
+> **CAUTION:** In this release, the Eventing component was switched completely from Knative to Kyma-internal Subscription custom resource (CR), which is based on the NATS server. Before upgrading to 1.21, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.21/docs/migration-guides/1.20-1.21.md) for details on Troubleshooting and Upgrade.
```
Comment (alexandra-simeonova) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:23]:
```suggestion
> **CAUTION:** In this release, the Eventing component was switched completely from Knative to NATS. Before upgrading to 1.21, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.21/docs/migration-guides/1.20-1.21.md) for details on Troubleshooting and Upgrade.
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 1 above)
Comment (alexandra-simeonova) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:68]:
```suggestion
For more information on the new Eventing, read the [documentation](https://kyma-project.io/docs/1.21/components/eventing).
```

**Review by alexandra-simeonova — COMMENTED** [review]
Great work! 🥇

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+We left the windy Amsterdam, but we're staying within the area of the West Germanic group of languages, this time stopping at Berlin, the capital of Germany and a world city of culture, politics, media, and science. 
+But don't be fooled by the great historical buildings, universities, and museums. 
+Berlin does a great job at combining this rich heritage with the new and avant-garde, and it does all that ahead of the rest, too. 
+Contemporary art, numerous festivals, artisanal coffee shops, craft beer, and the famous nightlife are just part of it. 
+Just like Berlin, Kyma also combines a solid foundation with innovation, providing high quality that will not cost you an arm and a leg. 
+This time, we focused on CLI, Monitoring, Eventing, and Serverless, so grab a cup of (artisanal) coffee, and read on.
+
+<!-- overview -->
+
+> **CAUTION:** In this release, the Eventing component was switched completely from Knative to NATS. Before upgrading to 1.21, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.21/docs/migration-guides/1.20-1.21.md) for details on Troubleshooting and Upgrade.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:23]: I'd say that the Migration Guide itself needs some more love 💔 But this comment doesn't concern this PR, of course.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+See the overview of all changes in this release:
+
+- [CLI](#cli) - alpha commands stabilized, `run` command available locally
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:27]:
```suggestion
- [CLI](#cli) - Alpha commands stabilized, the `run` command available locally
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+- [Monitoring](#monitoring) - dynamic datasource configuration, deriving traces from Loki logs
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:28]:
```suggestion
- [Monitoring](#monitoring) - Dynamic datasource configuration, deriving traces from Loki logs
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+- [Eventing](#eventing) - Knative-based Eventing replaced by NATS
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:29]:
```suggestion
- [Eventing](#eventing) - Knative-based eventing replaced by NATS 
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+- [Monitoring](#monitoring) - dynamic datasource configuration, deriving traces from Loki logs
+- [Eventing](#eventing) - Knative-based Eventing replaced by NATS 
+- [Serverless](#serverless) - Node.js 10 deprecation
+
+## CLI
+
+### New alpha commands stabilized 
+
+We've recently added new alpha commands: `alpha provision k3s`, `alpha deploy`, `alpha delete`, and  `alpha version`.
+In this release, we stabilized these commands, and they are now ready for beta-testing.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:37]:
```suggestion
In this release, we stabilized these commands, and they are now ready for beta testing.
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+For more information, read the [Alpha Command Usage Examples](https://kyma-project.io/docs/1.21/cli/overview#details-kyma-cli-alpha-command-usage-examples) and the command help.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:38]: Do we capitalize doc titles? Here I mean `Alpha Command Usage Examples` 🤔

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+We've recently added new alpha commands: `alpha provision k3s`, `alpha deploy`, `alpha delete`, and  `alpha version`.
+In this release, we stabilized these commands, and they are now ready for beta-testing.
+For more information, read the [Alpha Command Usage Examples](https://kyma-project.io/docs/1.21/cli/overview#details-kyma-cli-alpha-command-usage-examples) and the command help. 
+ 
+### Test-run Functions locally 
+
+Kyma CLI provides a set of helpful commands for Function developers.
+They support developers in starting development, pushing local code to the Runtime, and syncing Functions.
+In this release, we have added one more command, to facilitate your developer experience even more.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:44]:
```suggestion
In this release, we have added one more command to facilitate your developer experience even more.
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+## CLI
+
+### New alpha commands stabilized 
+
+We've recently added new alpha commands: `alpha provision k3s`, `alpha deploy`, `alpha delete`, and  `alpha version`.
+In this release, we stabilized these commands, and they are now ready for beta-testing.
+For more information, read the [Alpha Command Usage Examples](https://kyma-project.io/docs/1.21/cli/overview#details-kyma-cli-alpha-command-usage-examples) and the command help. 
+ 
+### Test-run Functions locally
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:40]: Can we use `test run` as a verb here? I've only found the noun examples (and without the hyphen). [Here](https://dictionary.cambridge.org/pl/dictionary/english/test-run) is the example. 

I think I'd just go with something like `Run Functions locally` or `New command to test Functions locally`.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+Kyma CLI provides a set of helpful commands for Function developers.
+They support developers in starting development, pushing local code to the Runtime, and syncing Functions.
+In this release, we have added one more command, to facilitate your developer experience even more.
+The `run` command allows you to run the Function's code locally, before applying it to the Kyma runtime.
+
+For more information, read the [Kyma CLI documentation](https://kyma-project.io/docs/1.21/cli/commands/#kyma-run-function-kyma-run-function). 
+
+## Monitoring
+
+### Dynamic data source configuration
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:51]: Datasource or data source? In the overview you use `datasource`, that's why I ask. The same goes for the whole paragraph below.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+The `run` command allows you to run the Function's code locally, before applying it to the Kyma runtime.
+
+For more information, read the [Kyma CLI documentation](https://kyma-project.io/docs/1.21/cli/commands/#kyma-run-function-kyma-run-function). 
+
+## Monitoring
+
+### Dynamic data source configuration 
+
+We enabled the Grafana feature to collect and load data source definitions from Kubernetes ConfigMaps at startup. 
+By doing that, we were able to move the individual pre-configured data sources into the related components. For example, the data source for Loki moved into the `logging` component. This way, we improved the resource organization.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:54]:
```suggestion
By doing that, we were able to move the individual pre-configured data sources into the related components. For example, the data source for Loki moved into the Logging component. This way, we improved the resource organization. 
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 7 above)
Comment (klaudiagrz) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:29]: In here we mean eventing as a general term, not the Eventing component, is that right? If that's the case, I'd use lowercase.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 13 above)
Comment (klaudiagrz) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:54]: I'd just capitalize Logging, just as we do for Application Connector, Service Catalog, and all the other components.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+The Loki data source for Grafana now contains a new derived field for `traceId`. 
+By default, Grafana checks every log in the Grafana View for the `traceId=(\w+)` pattern. 
+If it finds this `traceId`, it displays the link to the corresponding Jaeger trace.
+
+For more information about derived fields, read the [Grafana documentation](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
+
+## Eventing
+
+### Knative-based Eventing replaced by NATS
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:66]:
```suggestion
### Knative-based eventing replaced by NATS 
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+By default, Grafana checks every log in the Grafana View for the `traceId=(\w+)` pattern. 
+If it finds this `traceId`, it displays the link to the corresponding Jaeger trace.
+
+For more information about derived fields, read the [Grafana documentation](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
+
+## Eventing
+
+### Knative-based Eventing replaced by NATS 
+
+In this release, we removed Knative-based Eventing.
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:68]:
```suggestion
In this release, we removed Knative-based eventing. 
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+If it finds this `traceId`, it displays the link to the corresponding Jaeger trace.
+
+For more information about derived fields, read the [Grafana documentation](https://grafana.com/docs/grafana/latest/datasources/loki/#derived-fields).
+
+## Eventing
+
+### Knative-based Eventing replaced by NATS 
+
+In this release, we removed Knative-based Eventing. 
+The new Eventing is based on [NATS](https://github.com/nats-io/nats-server) with a Kyma-specific Subscription custom resource (CR).
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:69]:
```suggestion
The new eventing is based on [NATS](https://github.com/nats-io/nats-server) with a Kyma-specific Subscription custom resource (CR). 
```

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+
+## Eventing
+
+### Knative-based Eventing replaced by NATS 
+
+In this release, we removed Knative-based Eventing. 
+The new Eventing is based on [NATS](https://github.com/nats-io/nats-server) with a Kyma-specific Subscription custom resource (CR). 
+For upgrade scenarios, refer to the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.21/docs/migration-guides/1.20-1.21.md).
+
+For more information on the new Eventing, read the [documentation](https://kyma-project.io/docs/1.21/components/eventing).
```
Comment (klaudiagrz) [inline content/blog-posts/2021-03-18-release-notes-1.21/index.md:72]:
```suggestion
For more information on the new eventing, read the [documentation](https://kyma-project.io/docs/1.21/components/eventing).
```

**Review by klaudiagrz — COMMENTED** [review]
Also, remember to add borders to the screenshots 😄 

Cheers,
**Please-add-the-border** commando

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+  - "/blog/release-notes-121"
+---
+
+Herzlich Wilkommen! 
+We left the windy Amsterdam, but we're staying within the area of the West Germanic group of languages, this time stopping at Berlin, the capital of Germany and a world city of culture, politics, media, and science. 
+But don't be fooled by the great historical buildings, universities, and museums. 
+Berlin does a great job at combining this rich heritage with the new and avant-garde, and it does all that ahead of the rest, too. 
+Contemporary art, numerous festivals, artisanal coffee shops, craft beer, and the famous nightlife are just part of it. 
+Just like Berlin, Kyma also combines a solid foundation with innovation, providing high quality that will not cost you an arm and a leg. 
+This time, we focused on CLI, Monitoring, Eventing, and Serverless, so grab a cup of (artisanal) coffee, and read on.
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:19]: Thank you <3 I also have @NHingerl to help for input on Berlin 😊

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 7 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:29]: I think it's both, actually. The eventing in our Eventing changed, so our Eventing changed :D

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 9 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:38]: I took it as the doc title, but the title on the website is slightly different. I'll change it.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 11 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:40]: I see where you're coming from, but I believe the idea here is that this test run won't apply it to the Runtime, not that you can run it locally. 
@kazydek Can you provide more insight?

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 10 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:44]: I wanted to treat this as an addition, but I can also remove this comma.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 12 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:51]: In the provided input it was `datasource`, like in code, but Grafana and our docs spell it `data source`, so I changed that. Forgot about the intro, so thanks for noticing this!

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 13 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:54]: I wanted to put the emphasis on code here, but I can change that.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 5 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:27]: I'd skip the article here, as they are more like newspaper titles here.

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 7 above)
Comment (majakurcius) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:29]: I consulted @alexandra-simeonova to make sure what we focus on here. I'll change it to `eventing` :)

File: content/blog-posts/2021-03-18-release-notes-1.21/index.md (hunk 11 above)
Comment (kazydek) [reply content/blog-posts/2021-03-18-release-notes-1.21/index.md:40]: Let's go with `Run Functions locally` 👍

File: content/blog-posts/2021-04-06-release-notes-1.21/index.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+
+For more information on the new eventing, read the [documentation](https://kyma-project.io/docs/1.21/components/eventing).
+
+## Serverless 
+
+### Node.js 10 deprecation
+
+Node.js 10 release reaches end-of-life on Apr 30, 2021, and its base image will be deprecated.
+Following the recommendations from Node.js, we are planning to remove Node.js 10 from the list of supported runtimes **in the next release**. 
+This means that Node.js-10-based Functions will continue to work, but you won’t be able to create a new Function on it or edit the existing Function to run on it.
```
Comment (kazydek) [inline content/blog-posts/2021-04-06-release-notes-1.21/index.md:83]:
```suggestion

```

File: content/blog-posts/2021-04-06-release-notes-1.21/index.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+
+For now, we recommend that you configure all your Node.js 10 Functions to run on the Node.js 12 runtime.
```
Comment (kazydek) [inline content/blog-posts/2021-04-06-release-notes-1.21/index.md:85]:
```suggestion
We recommend that you configure all your Node.js 10 Functions to run on the Node.js 12 runtime. 
```

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**